src/main/java/graphql/util/FpKit.java

```diff
@@ -151,4 +152,12 @@
                 .collect(Collectors.toList());
     }
 
+    public static <T> T findOne(List<T> list, Predicate<T> filter) {
```

Comment:
Perhaps put the or else in the parameters.  More useable

```
  FpKit.findfirst(list, x -> isGood(x), otherwiseThis)
```

Could be a supplier to make it lazy

```
  FpKit.findfirst(list, x -> isGood(x), () -> otherwiseThis)
```

Reply:
or just return the optional and make the caller handle it.

```
    Object x = findFirst(list, this::isGood).orElse(null)
```
That is better to be honest

Reply:
changed it to fineOne AND findOneOrNull: our preference for returning null makes it a very common use case for us.